lib/chef/provider/package/chocolatey.rb

```diff
@@ -0,0 +1,190 @@
+#
+# Copyright:: Copyright (c) 2015 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/provider/package'
+require 'chef/resource/chocolatey_package'
+require 'chef/mixin/powershell_out'
+
+class Chef
+  class Provider
+    class Package
+      class Chocolatey < Chef::Provider::Package
+        include Chef::Mixin::PowershellOut
+
+        provides :chocolatey_package, os: "windows"
+
+        # Declare that our arguments should be arrays
+        package_class_supports_arrays
+
+        # Responsible for building the current_resource and as a (necessary)
+        # side effect this populates the candidate_version Array.
+        #
+        # @return [Chef::Resource::ChocolateyPackage] the current_resource
+        def load_current_resource
```

Comment:
Yeah we should probably do a check for choco.exe existence.  Its really fussy getting that right, though.

I feel its now necessary to to populate candidate_version in load_current_resource for inspecting with `current_value`.  I'm having some active discussion with @jkeiser about what to do about 'derived desired state' like this and playing around with ways to get at provider state off of `current_value`.  Currently, we don't call `define_resource_requirements` or `process_resource_requirements` in `current_value`, though, so the right place to set `candidate_version` seems to be here.